taskcluster/ci/test/kind.yml

```diff
@@ -0,0 +1,50 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+job-defaults:
+    description: Test Reference Browser
+    treeherder:
+        kind: test
+        tier: 2
+    worker-type: b-android
+    worker:
+        max-run-time: 7200
+    run:
+        using: gradlew
+        use-caches: false
+
+jobs:
+    ui:
+        attributes:
+            build-type: debug
+            code-review: true
+        treeherder:
+            symbol: ui
+            platform: 'ui-test/opt'
+        run-on-tasks-for: [github-pull-request, github-push]
+        run:
+            # TODO Generate APKs in a build task instead
```

Comment:
I don't quite follow - why do we want to generate APKs in a separate build task?

Reply:
We have some duplication of builds here. `assembleDebug` is done in the PR build task. That's not entirely true, because we just `assembleNightlyDebug` there, but the UI tests are just interested in the nightly build anymore.

While this debug build is assembled, we can also parallelize `assembleAndroidTest`. This way, the UI tests just run the tests, and should take less time to execute overall. I heard the duration of the UI tests is becoming a pain point in PRs.